crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -211,10 +201,13 @@ impl MultiFarming {
         let archiving = Archiving::start(farmer_metadata, object_mappings, client.clone(), {
             let mut on_pieces_to_plots = plots
                 .iter()
-                .zip(subspace_codecs)
                 .zip(&commitments)
-                .map(|((plot, subspace_codec), commitments)| {
-                    plotting::plot_pieces(subspace_codec, plot, commitments.clone())
+                .map(|(plot, commitments)| {
+                    plotting::plot_pieces(
+                        SubspaceCodec::new(&plot.public_key()),
```

Comment:
No, instead of inlining codec creation we should have one instance used everywhere, there were TODOs about this in the past where we had extra instances created unnecessarily, so the opposite of this commit needs to be done.

Reply:
Will it be fine to share those under `Arc<Mutex>` between various threads?

Reply:
No need to wrap them, it is a copy type right now and should remain clonable going forward. Just replicate its copies as much as needed, but don't call constructor all the time since that'll be more expensive (especially when GPU is used).